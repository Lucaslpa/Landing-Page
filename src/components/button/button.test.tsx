import { render } from 'utils/test-utils';
import { screen } from '@testing-library/react';
import { Button } from './index';

describe.only('Button', () => {
  it('Should render small button dark', () => {
    render(<Button size="small" colors="dark" label="button" />);
    const button = screen.getByLabelText(/button/gi);
    expect(button).toHaveStyleRule('color', '#FFFFFF');
    expect(button).toHaveStyleRule('background-color', '#252160');
    expect(button).toHaveStyleRule('fill', '#FFFFFF');
    expect(button).toHaveStyleRule('font-size', '1.5rem');
    expect(button).toHaveStyleRule('max-height', '4.3rem');
    expect(button).toHaveStyleRule('max-width', '13.3rem');
    expect(button).toMatchSnapshot();
  });

  it('Should render medium button dark', () => {
    render(<Button size="medium" colors="dark" label="button" />);
    const button = screen.getByLabelText(/button/gi);
    expect(button).toHaveStyleRule('color', '#FFFFFF');
    expect(button).toHaveStyleRule('background-color', '#252160');
    expect(button).toHaveStyleRule('fill', '#FFFFFF');
    expect(button).toHaveStyleRule('font-size', '1.7rem');
    expect(button).toHaveStyleRule('max-height', '6rem');
    expect(button).toHaveStyleRule('max-width', '20rem');

    expect(button).toMatchSnapshot();
  });

  it('Should render medium button ligh', () => {
    render(<Button size="medium" colors="light" label="button" />);
    const button = screen.getByLabelText(/button/gi);
    expect(button).toHaveStyleRule('color', '#252160');
    expect(button).toHaveStyleRule('background-color', '#3AFFE5');
    expect(button).toHaveStyleRule('fill', '#252160');
    expect(button).toHaveStyleRule('font-size', '1.7rem');
    expect(button).toHaveStyleRule('max-height', '6rem');
    expect(button).toHaveStyleRule('max-width', '20rem');
    expect(button).toMatchSnapshot();
  });

  it('Should render big button lightGradient', () => {
    render(<Button size="big" colors="lightGradient" label="button" />);
    const button = screen.getByLabelText(/button/gi);
    expect(button).toHaveStyleRule('color', '#FFFFFF');
    expect(button).toHaveStyleRule(
      'background',
      'linear-gradient(145.09deg,#26E9CF 11.81%,#25CDE1 95.89%)'
    );
    expect(button).toHaveStyleRule('fill', '#FFFFFF');
    expect(button).toHaveStyleRule('font-size', '1.7rem');
    expect(button).toHaveStyleRule('max-height', '6rem');
    expect(button).toHaveStyleRule('max-width', '31.3rem');

    expect(button).toMatchSnapshot();
  });
  it('Should render xBig button ', () => {
    render(<Button size="xBig" colors="lightGradient" label="button" />);
    const button = screen.getByLabelText(/button/gi);
    expect(button).toHaveStyleRule('font-size', '1.7rem');
    expect(button).toHaveStyleRule('max-height', '5.9rem');
    expect(button).toHaveStyleRule('max-width', '38rem');
    expect(button).toMatchSnapshot();
  });
  it('Should render button with arrow icon', () => {
    render(<Button size="small" colors="dark" label="button" ArrowRight />);
    const arrow = screen.getByLabelText(/arrow/gi);
    expect(arrow).toBeInTheDocument();
    expect(arrow).toMatchSnapshot();
  });
  it('Should render button without arrow icon', () => {
    render(<Button size="small" colors="dark" label="button" />);
    const arrow = screen.queryByLabelText(/arrow/gi);
    expect(arrow).not.toBeInTheDocument();
    expect(arrow).toMatchSnapshot();
  });
});
