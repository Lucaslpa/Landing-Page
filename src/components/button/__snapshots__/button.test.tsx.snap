// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Should render big button lightGradient 1`] = `
.c0 {
  cursor: pointer;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  box-sizing: border-box;
  height: 100vh;
  width: 100%;
  cursor: pointer;
  font-family: Inter;
  font-weight: bold;
  line-height: 21px;
  font-style: normal;
  color: #FFFFFF;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  fill: #FFFFFF;
  background: linear-gradient(145.09deg,#26E9CF 11.81%,#25CDE1 95.89%);
  max-width: 31.3rem;
  max-height: 6rem;
  font-size: 1.7rem;
  text-align: center;
  border-radius: 0.6rem;
}

.c0 path {
  fill: #FFFFFF;
}

.c0:hover {
  background: linear-gradient(145.09deg,#22bfaa 11.81%,#25cde1 95.89%);
}

@media (max-width:768px) {
  .c0 {
    font-size: 1.5rem;
    max-width: 29.4rem;
    max-height: 5.7rem;
  }

  .c0 svg {
    width: 1.7rem;
  }
}

<a
  aria-label="button"
  class="c0"
  href="/"
>
  <span>
    Button
  </span>
</a>
`;

exports[`Button Should render button with arrow icon 1`] = `
<svg
  aria-label="arrow"
  fill="none"
  height="17"
  viewBox="0 0 20 17"
  width="20"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M2.02165 9.89163H15.8188L11.5959 14.9638C11.4981 15.0814 11.4245 15.2171 11.3791 15.3632C11.3338 15.5093 11.3177 15.6629 11.3318 15.8152C11.3601 16.1229 11.5095 16.4066 11.7471 16.6041C11.9847 16.8015 12.291 16.8965 12.5986 16.8682C12.9062 16.8398 13.19 16.6904 13.3874 16.4528L19.2041 9.47283C19.2432 9.41731 19.2782 9.35899 19.3088 9.29833C19.3088 9.24016 19.3669 9.20526 19.3902 9.1471C19.4429 9.01371 19.4706 8.87172 19.4716 8.7283C19.4706 8.58487 19.4429 8.44288 19.3902 8.3095C19.3902 8.25133 19.332 8.21643 19.3088 8.15826C19.2782 8.09761 19.2432 8.03928 19.2041 7.98376L13.3874 1.00376C13.278 0.872443 13.1411 0.766836 12.9862 0.694452C12.8314 0.622068 12.6626 0.584686 12.4916 0.584963C12.2198 0.584432 11.9564 0.679099 11.7471 0.85253C11.6293 0.950191 11.5319 1.07013 11.4606 1.20548C11.3892 1.34083 11.3452 1.48893 11.3312 1.6413C11.3171 1.79367 11.3333 1.94732 11.3787 2.09344C11.4241 2.23955 11.4979 2.37528 11.5959 2.49283L15.8188 7.56496H2.02165C1.71311 7.56496 1.41721 7.68753 1.19905 7.9057C0.98088 8.12386 0.858315 8.41976 0.858315 8.7283C0.858315 9.03683 0.98088 9.33273 1.19905 9.5509C1.41721 9.76906 1.71311 9.89163 2.02165 9.89163Z"
    fill="#252160"
  />
</svg>
`;

exports[`Button Should render button without arrow icon 1`] = `null`;

exports[`Button Should render medium button dark 1`] = `
.c0 {
  cursor: pointer;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  box-sizing: border-box;
  height: 100vh;
  width: 100%;
  cursor: pointer;
  font-family: Inter;
  font-weight: bold;
  line-height: 21px;
  font-style: normal;
  color: #FFFFFF;
  background-color: #252160;
  fill: #FFFFFF;
  max-width: 20rem;
  max-height: 6rem;
  font-size: 1.7rem;
  text-align: center;
  border-radius: 0.6rem;
}

.c0 path {
  fill: #FFFFFF;
}

.c0:hover {
  background-color: #27216e;
}

@media (max-width:768px) {
  .c0 {
    font-size: 1.3rem;
    max-width: 11.1rem;
    max-height: 3.3rem;
  }

  .c0 svg {
    width: 1.7rem;
  }
}

<a
  aria-label="button"
  class="c0"
  href="/"
>
  <span>
    Button
  </span>
</a>
`;

exports[`Button Should render medium button ligh 1`] = `
.c0 {
  cursor: pointer;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  box-sizing: border-box;
  height: 100vh;
  width: 100%;
  cursor: pointer;
  font-family: Inter;
  font-weight: bold;
  line-height: 21px;
  font-style: normal;
  color: #252160;
  background-color: #3AFFE5;
  fill: #252160;
  max-width: 20rem;
  max-height: 6rem;
  font-size: 1.7rem;
  text-align: center;
  border-radius: 0.6rem;
}

.c0 path {
  fill: #252160;
}

.c0:hover {
  background-color: #32e3cc;
}

@media (max-width:768px) {
  .c0 {
    font-size: 1.3rem;
    max-width: 11.1rem;
    max-height: 3.3rem;
  }

  .c0 svg {
    width: 1.7rem;
  }
}

<a
  aria-label="button"
  class="c0"
  href="/"
>
  <span>
    Button
  </span>
</a>
`;

exports[`Button Should render small button dark 1`] = `
.c0 {
  cursor: pointer;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  box-sizing: border-box;
  height: 100vh;
  width: 100%;
  cursor: pointer;
  font-family: Inter;
  font-weight: bold;
  line-height: 21px;
  font-style: normal;
  color: #FFFFFF;
  background-color: #252160;
  fill: #FFFFFF;
  max-width: 13.3rem;
  max-height: 4.3rem;
  font-size: 1.5rem;
  text-align: center;
  border-radius: 0.6rem;
}

.c0 path {
  fill: #FFFFFF;
}

.c0:hover {
  background-color: #27216e;
}

<a
  aria-label="button"
  class="c0"
  href="/"
>
  <span>
    Button
  </span>
</a>
`;

exports[`Button Should render xBig button  1`] = `
.c0 {
  cursor: pointer;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-evenly;
  -webkit-justify-content: space-evenly;
  -ms-flex-pack: space-evenly;
  justify-content: space-evenly;
  box-sizing: border-box;
  height: 100vh;
  width: 100%;
  cursor: pointer;
  font-family: Inter;
  font-weight: bold;
  line-height: 21px;
  font-style: normal;
  color: #FFFFFF;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  fill: #FFFFFF;
  background: linear-gradient(145.09deg,#26E9CF 11.81%,#25CDE1 95.89%);
  max-width: 38rem;
  max-height: 5.9rem;
  font-size: 1.7rem;
  text-align: center;
  border-radius: 0.6rem;
}

.c0 path {
  fill: #FFFFFF;
}

.c0:hover {
  background: linear-gradient(145.09deg,#22bfaa 11.81%,#25cde1 95.89%);
}

@media (max-width:768px) {
  .c0 {
    font-size: 1.5rem;
    max-width: 29.4rem;
    max-height: 4.6rem;
  }

  .c0 svg {
    width: 1.4rem;
  }
}

<a
  aria-label="button"
  class="c0"
  href="/"
>
  <span>
    Button
  </span>
</a>
`;
